src/Operator/Binary/AbstractBinaryOperator.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Operator\Binary;
+
+use Twig\Operator\AbstractOperator;
+use Twig\Operator\OperatorArity;
+use Twig\Operator\OperatorAssociativity;
+
+abstract class AbstractBinaryOperator extends AbstractOperator
+{
+    public function getArity(): OperatorArity
+    {
+        return OperatorArity::Binary;
+    }
+
+    public function getAssociativity(): OperatorAssociativity
+    {
+        return OperatorAssociativity::Left;
+    }
+
+    public function getCallable(): ?callable
```

Comment:
this should have a phpdoc `@return (callable(ExpressionParser, AbstractExpression): AbstractExpression)|null` to describe the expected callable signature (putting this phpdoc on the highest level defining the method, which is currently `AbstractBinaryOperator` but might be the interface if you add it in an interface)